installer/uninstaller.sh

```diff
@@ -38,30 +38,60 @@ CHARTS_LIST="$RELEASE_PREFIX-mgt-api $RELEASE_PREFIX-crd $RELEASE_PREFIX-dex $RE
 HELM_LS_OUTPUT=`helm ls --output json`
 HELM_LIST=`jq --arg namearg "Releases" '.[$namearg]' <<< $HELM_LS_OUTPUT`
 K8S_NS_OUTPUT=`kubectl get ns --output=json`
-K8S_NS_LIST=`jq --arg namearg "items" '.[$namearg]' <<< $K8S_NS_OUTPUT`
+K8S_NAMESPACES=`echo $K8S_NS_OUTPUT | jq '.items[].metadata.name'`
 helm_arr=()
 K8S_NS_ARR=()
 
+[[ $K8S_NAMESPACES =~ "default-tenant" ]] && SINGLE_TENANT=1 || SINGLE_TENANT=0
+
 cd ../scripts
-./imm -k rm t default-tenant
-cd ../installer
+if [[ "$quiet" != "yes" ]]; then
+    echo "Uninstalling IMM. Following components will be removed:"
+    echo "*Helm releases (mgt-api, crd etc.)"
+    if [[ $SINGLE_TENANT == 1 ]]; then
+        echo "*Default tenant with it's resources(models,endpoints etc.)"
+    fi
+    echo "Are you sure you want to uninstall IMM? Y/n"
+    read DELETE_IMM
+else
+    DELETE_IMM="Y"
+fi
+if [[ $DELETE_IMM == "Y" ]]; then
 
-while read i; do
-   echo $i
-   RELEASE_NAME=`jq --arg namearg "Name" '.[$namearg]' <<< $i | tr -d '"'`
-   mark_release_to_be_deleted $RELEASE_NAME
-done <<<"$(jq -c '.[]' <<< $HELM_LIST)"
+    if [[ $SINGLE_TENANT == 1 ]]; then
+        echo "Deleting default-tenant..."
+        RESULT=`./imm -k rm t default-tenant`
+    else
+        RESULT=0
+    fi
 
+    if [[ "$RESULT" =~ "Tenant default-tenant deleted" || $RESULT == 0 ]]; then
+        cd ../installer
 
-echo "Releases marked to be deleted:"
+        while read i; do
+           echo $i
+           RELEASE_NAME=`jq --arg namearg "Name" '.[$namearg]' <<< $i | tr -d '"'`
+           mark_release_to_be_deleted $RELEASE_NAME
+        done <<<"$(jq -c '.[]' <<< $HELM_LIST)"
 
-for i in "${helm_arr[@]}"; do echo "$i" ; done
+        echo "Releases marked to be deleted:"
 
-if [[ "$quiet" == "yes" ]]; then
-    for i in "${helm_arr[@]}"; do helm del --debug --purge $i ; done
+        for i in "${helm_arr[@]}"; do echo "$i" ; done
+
+        if [[ "$quiet" == "yes" ]]; then
+            for i in "${helm_arr[@]}"; do helm del --debug --purge $i ; done
+        else
+            read -p "Do you want to delete helm releases listed above Y/n?" DELETE_CHARTS
+            [[ $DELETE_CHARTS != "n" ]] && for i in "${helm_arr[@]}"; do helm del --debug --purge $i ; done
+        fi
+
+        kubectl delete ing minio-ingress || true
+        echo "IMM uninstalled successfully."
+    elif [[ "$RESULT" =~ "Token not valid" ]]; then
```

Comment:
i think it would be better to first check the RESULT of deleting default tenant and in `else` continue wit uninstalling.. Because here when I was reading this code I already forget what was the variable RESULT responsible for.

Reply:
RESULT is mentioned both in `if` and `elif`, so maybe changing variable name into something more indicative  would be better solution?